### ISSUE TITLE
Crash when typing “he” in to field

### ISSUE THREAD

**Issue by rol1** [thunderbird/thunderbird-android#5766]
Hi,

when I try to send an e-mail to a receiver who’s name starts with “he” the create new mail window closes, and I'm back on the K9 shared inbox. I can reproduce this since several weeks.
![Screenshot_2021-11-05-14-01-01-940_com fsck k9](https://user-images.githubusercontent.com/6573464/140515601-360465e9-0353-4d2d-91c5-e303fbfcccf4.jpg)

K9 Version: 5.806
Android Version: 11 RP1A200720.012
Xiaomi MIUI Version: 12.5.3

Yours

Roland

**Comment by rol1** [comment]
Today I got an update of Xiaomi MIUI. No I have MIUI-V12.5.4.0.RGGEUXM
I just noticed a typo in the Android version, it is 11 RP1A.200720.011
K9 version is still 5.806

The behavior, that the Mail "window" closes when I type "he" into the to field is still the same.

**Comment by cketti** [comment]
Please provide a debug log. See https://github.com/k9mail/k-9/wiki/LoggingErrors

**Comment by rol1** [comment]
mhh, adb devices does not show my phone ...

**Comment by rol1** [comment]
On thing I noticed while debugging was activated was, that K9 was closed completly. Without debugging only the "new mail window" gets closed, and I'm back on the shared inbox.

**Comment by rol1** [comment]
Finaly I can provide the debug log
[k9-log.txt](https://github.com/k9mail/k-9/files/7595105/k9-log.txt)

**Comment by cketti** [comment]
Thanks for the debug log :+1:

Exception from the log:
```
java.lang.RuntimeException: An error occurred while executing doInBackground()
	at androidx.loader.content.ModernAsyncTask$3.done(ModernAsyncTask.java:164)
	at java.util.concurrent.FutureTask.finishCompletion(FutureTask.java:383)
	at java.util.concurrent.FutureTask.setException(FutureTask.java:252)
	at java.util.concurrent.FutureTask.run(FutureTask.java:271)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1167)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:641)
	at java.lang.Thread.run(Thread.java:923)
Caused by: java.lang.IllegalArgumentException: Comparison method violates its general contract!
	at java.util.TimSort.mergeHi(TimSort.java:899)
	at java.util.TimSort.mergeAt(TimSort.java:516)
	at java.util.TimSort.mergeCollapse(TimSort.java:441)
	at java.util.TimSort.sort(TimSort.java:245)
	at java.util.Arrays.sort(Arrays.java:1492)
	at java.util.ArrayList.sort(ArrayList.java:1470)
	at java.util.Collections.sort(Collections.java:206)
	at com.fsck.k9.activity.compose.RecipientLoader.fillContactDataFromQuery(RecipientLoader.java:342)
	at com.fsck.k9.activity.compose.RecipientLoader.loadInBackground(RecipientLoader.java:195)
	at com.fsck.k9.activity.compose.RecipientLoader.loadInBackground(RecipientLoader.java:37)
	at androidx.loader.content.AsyncTaskLoader.onLoadInBackground(AsyncTaskLoader.java:307)
	at androidx.loader.content.AsyncTaskLoader$LoadTask.doInBackground(AsyncTaskLoader.java:60)
	at androidx.loader.content.AsyncTaskLoader$LoadTask.doInBackground(AsyncTaskLoader.java:48)
	at androidx.loader.content.ModernAsyncTask$2.call(ModernAsyncTask.java:141)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	... 3 more
```

**Comment by rol1** [comment]
I already thought that an error occured ...

**Comment by rol1** [comment]
Thank you very much dir the quick response
Can you estimate when this fix com into a News version of K9?

**Comment by cketti** [comment]
The fix is included in the latest beta version: [K-9 Mail 5.905](https://github.com/k9mail/k-9/releases/tag/5.905)

**Comment by rol1** [comment]
thanks
